lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
+          }
 
-      if (!this.types[field.type].validate(field.typeOptions, fieldValues[i], fieldErrors)) {
-        if (fieldErrors.length === 0) {
-          // We still want to trigger an error, even if no message is provided
-          manageErrorMessage(field.path, errorMessages, 'An error has occurred during validation.', verbose);
-        }
-        else {
-          fieldErrors.forEach(message => manageErrorMessage(field.path, errorMessages, message, verbose));
+          return false;
         }
 
-        return false;
-      }
+        if (this.types[field.type].allowChildren && field.children) {
+          try {
+            if (!this.recurseFieldValidation(fieldValues[i], field.children, nestedStrictness, errorMessages, verbose)) {
+              result = false;
+            }
+          }
+          catch (error) {
+            if (error.message === 'strictness') {
+              manageErrorMessage(field.path, errorMessages, 'The field is set to "strict"; cannot add unspecified subField.', verbose);
+            }
+            else if (verbose) {
+              manageErrorMessage(field.path, errorMessages, error.message, verbose);
+            }
+            else {
+              throw error;
+            }
 
-      if (this.types[field.type].allowChildren && field.children) {
-        try {
-          if (!this.recurseFieldValidation(fieldValues[i], field.children, nestedStrictness, errorMessages, verbose)) {
             result = false;
           }
         }
-        catch (error) {
-          if (error.message === 'strictness') {
-            manageErrorMessage(field.path, errorMessages, 'The field is set to "strict"; cannot add unspecified subField.', verbose);
-          }
-          else if (verbose) {
-            manageErrorMessage(field.path, errorMessages, error.message, verbose);
-          }
-          else {
-            throw error;
-          }
-
-          result = false;
-        }
       }
     }
+
+    return result;
   }
 
-  return result;
-};
+  /**
+   * @returns {Promise.<T>}
+   */
+  curateSpecification() {
+    var
+      promises = [],
+      specification = {};
 
-/**
- * @returns {Promise.<T>}
- */
-Validation.prototype.curateSpecification = function validationCurateSpecification () {
-  var
-    promises = [],
-    specification = {};
+    return getValidationConfiguration(this.kuzzle)
+      .then(validation => {
+        this.rawConfiguration = validation;
 
-  return getValidationConfiguration(this.kuzzle)
-    .then(validation => {
-      this.rawConfiguration = validation;
+        Object.keys(this.rawConfiguration).forEach(indexName => {
+          Object.keys(this.rawConfiguration[indexName]).forEach(collectionName => {
+            var promise = new Bluebird((resolve) => {
+              this.curateCollectionSpecification(indexName, collectionName, this.rawConfiguration[indexName][collectionName])
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 125 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)